File: LeetCode/104MaximumDepthOfBinaryTree/MaximumDepthOfBinaryTree.cpp
class Solution {
public:
    int maxDepth(TreeNode* root) {
        // Inorder traversal -> Time:O(n), Space:O(1)
        if (root == NULL) {
            return 0;
        }
        return (1 + max(maxDepth(root->left), maxDepth(root->right)));
    }
};

File: LeetCode/104MaximumDepthOfBinaryTree/MaximumDepthOfBinaryTree.py
class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        # Inorder traversal -> Time:O(n), Space:O(1)
        if root == None:
            return 0
        else:
            return 1 + max(self.maxDepth(root.left), self.maxDepth(root.right))
